go/cbrotli.go

```diff
@@ -0,0 +1,264 @@
+// Copyright 2016 Google Inc. All Rights Reserved.
+//
+// Distributed under MIT license.
+// See file LICENSE for detail or copy at https://opensource.org/licenses/MIT
+
+// Package cbrotli compresses and decompresses data with C-Brotli library.
+package cbrotli
+
+import (
+	"bytes"
+	"errors"
+	"io"
+
+	"github.com/google/brotli/go/cgo/decoder"
+	"github.com/google/brotli/go/cgo/encoder"
+)
+
+//------------------------------------------------------------------------------
+// Encoder
+//------------------------------------------------------------------------------
+
+// WriterOptions configures Writer.
+type WriterOptions struct {
+	// Quality controls the compression-speed vs compression-density trade-offs.
+	// The higher the quality, the slower the compression. Range is 0 to 11.
+	Quality int
+	// LGWin is the base 2 logarithm of the sliding window size.
+	// Range is 10 to 24. 0 indicates automatic configuration based on Quality.
+	LGWin int
+}
+
+// ErrEncodeFailure is returned when the encode operation fails.
+var ErrEncodeFailure = errors.New("Brotli encode failure")
+
+// ErrWriteAfterClose is returned from Write() if called after Close().
+var ErrWriteAfterClose = errors.New("brotli.Writer: write after close")
```

Comment:
Unexport this. I know the [`archive/tar.ErrWriteAfterClose`](https://golang.org/pkg/archive/tar/#pkg-variables) does this, but I consider that a mistake in retrospect.